### ISSUE TITLE
msgpack's Session assumes it's not executed from the main greenlet

### ISSUE THREAD

**Issue by hackaugusto** [neovim/pynvim#355]
This is the full stacktrace, I guess the first two errors are not related:

```
Error detected while processing function <SNR>77_OnCursorMovedNormalMode[1]..<SNR>77_AllowedToCompleteInCurrentBuffer[1]..<
SNR>77_AllowedToCompleteInBuffer:
line   15:
E715: Dictionary required
Error detected while processing function <SNR>77_OnCursorMovedNormalMode:
line    5:
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/usr/share/vim/vimfiles/autoload/../python/ycm/client/base_request.py", line 193, in Session
    return cls.session
AttributeError: type object 'BaseRequest' has no attribute 'session'
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/share/vim/vimfiles/autoload/../python/ycm/client/ycmd_keepalive.py", line 48, in _ThreadMain
    BaseRequest().GetDataFromHandler( 'healthy', display_message = False )
  File "/usr/share/vim/vimfiles/autoload/../python/ycm/client/base_request.py", line 110, in GetDataFromHandler
    BaseRequest._TalkToHandlerAsync( '', handler, 'GET', timeout ),
  File "/usr/share/vim/vimfiles/autoload/../python/ycm/client/base_request.py", line 159, in _TalkToHandlerAsync
    return BaseRequest.Session().get(
  File "/usr/share/vim/vimfiles/autoload/../python/ycm/client/base_request.py", line 196, in Session
    from requests_futures.sessions import FuturesSession
  File "/usr/share/vim/vimfiles/third_party/requests-futures/requests_futures/sessions.py", line 24, in <module>
    from requests import Session
  File "/usr/share/vim/vimfiles/third_party/ycmd/third_party/requests/requests/__init__.py", line 53, in <module>
    from .packages.urllib3.contrib import pyopenssl
  File "/usr/share/vim/vimfiles/third_party/ycmd/third_party/requests/requests/packages/urllib3/contrib/pyopenssl.py", line 49, in <module>
    from ndg.httpsclient.ssl_peer_verification import SUBJ_ALT_NAME_SUPPORT
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 963, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 906, in _find_spec
  File "<frozen importlib._bootstrap_external>", line 1280, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1252, in _get_spec
  File "/usr/lib/python3.7/site-packages/neovim/plugin/script_host.py", line 247, in find_spec
    return PathFinder.find_spec(fullname, path or _get_paths(), target)
  File "/usr/lib/python3.7/site-packages/neovim/plugin/script_host.py", line 211, in _get_paths
    return discover_runtime_directories(nvim)
  File "/usr/lib/python3.7/site-packages/neovim/plugin/script_host.py", line 260, in discover_runtime_directories
    for rtp in nvim.list_runtime_paths():
  File "/usr/lib/python3.7/site-packages/neovim/api/nvim.py", line 316, in list_runtime_paths
    return self.request('nvim_list_runtime_paths')
  File "/usr/lib/python3.7/site-packages/neovim/api/nvim.py", line 170, in request
   res = self._session.request(name, *args, **kwargs)
  File "/usr/lib/python3.7/site-packages/neovim/msgpack_rpc/session.py", line 91, in request
    v = self._yielding_request(method, args)
  File "/usr/lib/python3.7/site-packages/neovim/msgpack_rpc/session.py", line 160, in _yielding_request
    return parent.switch()
AttributeError: 'NoneType' object has no attribute 'switch'
```

I'm running Archlinux, these are the package versions:

python version: 3.7.0
neovim version: 0.3.1
python3 neovim version: 0.2.6
vim-youcompleteme-git version: 2373.15362d9c-1

**Comment by justinmk** [comment]
Can you suggest a fix ?

**Comment by bfredl** [comment]
#255 might fix this.

**Comment by justinmk** [comment]
closed by #255
